tutorials/xr/openxr_spatial_entities.rst

```diff
@@ -0,0 +1,1738 @@
+        if discovery_result:
+            discovery_result.cancel_discovery()
+
+        # Perform our discovery.
+        discovery_result = OpenXRSpatialEntityExtension.discover_spatial_entities(spatial_context, [ \
+                OpenXRSpatialEntityExtension.COMPONENT_TYPE_BOUNDED_2D, \
+                OpenXRSpatialEntityExtension.COMPONENT_TYPE_PLANE_ALIGNMENT \
+            ])
+
+        # Wait for async completion.
+        await discovery_result.completed
+
+        var snapshot : RID = discovery_result.get_spatial_snapshot()
+        if snapshot:
+            # Process our snapshot result.
+            _process_snapshot(snapshot)
+
+            # And clean up our snapshot.
+            OpenXRSpatialEntityExtension.free_spatial_snapshot(snapshot)
+
+
+    func _process_snapshot(p_snapshot):
+        # See further down.
+        pass
+
+
+Note that when calling ``discover_spatial_entities`` we specify a list of components.
+The discovery query will find any entity that is managed by the spatial context and has
+at least one of the specified components.
+
+Update snapshots
+""""""""""""""""
+
+Performing an update snapshot allows us to get updated information about entities
+we already found previously with our discovery snapshot.
+This function is synchronous, and is mainly meant to obtain status and positioning data
+and can be run every frame.
+
+Generally speaking you would only perform update snapshots when it's likely entities
+change or have a lifetime process. A good example of this are persistent anchors and
+markers. Consult the documentation about a capability to determine if this is needed.
+
+It is not needed for plane tracking however to complete our example, here is an example
+of what an update snapshot would look like for plane tracking if we needed one:
+
+.. code-block:: gdscript
+
+    ...
+
+    func _process(_delta):
+        if not spatial_context:
+            return
+
+        var entity_rids : Array[RID]
+        for entity_id in entities:
+            entity_rids.push_back(entities[entity_id].entity)
+
+        var snapshot : RID = OpenXRSpatialEntityExtension.update_spatial_entities(spatial_context, entity_rids, [ \
+                OpenXRSpatialEntityExtension.COMPONENT_TYPE_BOUNDED_2D, \
+                OpenXRSpatialEntityExtension.COMPONENT_TYPE_PLANE_ALIGNMENT \
+            ])
+        if snapshot:
+            # Process our snapshot.
+            _process_snapshot(snapshot)
+
+            # And clean up our snapshot.
+            OpenXRSpatialEntityExtension.free_spatial_snapshot(snapshot)
+
+Note that in our example here we're using the same ``_process_snapshot`` function to process the snapshot.
+This makes sense in most situations. However if the components you've specified when creating the snapshot
+are different between your discovery snapshot and your update snapshot, you have to take that you query
+different components into account.
```

Comment:
I think some words got jumbled in this text: "you have to take that you query different components into account"